packages/block-editor/src/components/inserter/pattern-panel.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * WordPress dependencies
+ */
+import { CustomSelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+function PatternInserterPanel( {
+	selectedCategory,
+	patternCategories,
+	onClickCategory,
+	children,
+} ) {
+	const categoryOptions = () => {
+		const options = [];
+
+		patternCategories.map( ( patternCategory ) => {
+			return options.push( {
+				key: patternCategory.name,
+				name: patternCategory.label,
+				style: {},
+				className:
+					patternCategory.name === selectedCategory.name
+						? 'is-selected'
+						: '',
+			} );
+		} );
+
+		return options;
+	};
+
+	const onChangeSelect = ( selected ) => {
+		onClickCategory(
+			patternCategories.find(
+				( patternCategory ) =>
+					selected.selectedItem.key === patternCategory.name
+			)
+		);
+	};
+
+	return (
+		<>
+			<div className="block-editor-inserter__panel-header">
+				<CustomSelectControl
+					label={ __( 'Select a Category' ) }
```

Comment:
There is still a label "Select a Category" that is readable by screen readers, just not visible on the page.

Reply:
See: https://developer.wordpress.org/block-editor/components/custom-select-control/#hidelabelfromvision

Reply:
Visible labels should always be used when possible. `aria-label`s/tooltips should be a last resort.

>While the efforts towards accessibility have been significant, there has been a strong dependence on methods we consider to be anti-patterns. First among these is a heavy use of ARIA attributes. ARIA, while sometimes necessary, should be considered only as a final option when no native HTML interactions are available.

https://make.wordpress.org/accessibility/2018/10/29/report-on-the-accessibility-status-of-gutenberg/

Reply:
In this instance, although `aria-label="Select a Category"` is set on the button we are not relying on this solely as there is also a native HTML `<label>Select a Category</label>`  visible to screen readers. I think it is an acceptable practice to [hide the labels from the visual UI](https://www.w3.org/WAI/tutorials/forms/labels/#hiding-label-text).

I agree with @joanrho and @apeatling that the dropdown down-arrow is sufficient here in communicating the intent  in terms of the visual UI.